Frameworks/UIKit/UIView.mm

```diff
@@ -274,30 +260,106 @@ - (bool)_processGesturesForTouch:(UITouch*)touch event:(UIEvent*)event touchEven
         }
     }
 
-    // gesture priority list
-    const static id s_gesturesPriority[] = {[UIPinchGestureRecognizer class], [UISwipeGestureRecognizer class],
-                                            [UIPanGestureRecognizer class],   [UILongPressGestureRecognizer class],
-                                            [UITapGestureRecognizer class],   [_UIDMPanGestureRecognizer class] };
+    //
+    // After state transtion, we potentially get active gestures, we should remove any gestures that can not be recognized simultaneously
+    //
+    NSMutableSet* gesturesToRemove = [NSMutableSet set];
+    for (UIGestureRecognizer* curGesture in [g_currentlyTrackingGesturesList reverseObjectEnumerator]) {
+        UIGestureRecognizerState state = curGesture.state;
+        if (state != UIGestureRecognizerStatePossible && state != UIGestureRecognizerStateCancelled) {
+            UIGestureRecognizer* activeGesture = curGesture;
 
-    const static int s_numGestureTypes = sizeof(s_gesturesPriority) / sizeof(s_gesturesPriority[0]);
+            BOOL activeResponds = [activeGesture.delegate
+                respondsToSelector:@selector(gestureRecognizer:shouldRecognizeSimultaneouslyWithGestureRecognizer:)];
 
-    //  Process all gestures, including DM gesture
-    for (auto const& curgestureClass : s_gesturesPriority) {
-        id gestures = [g_curGesturesDict objectForKey:curgestureClass];
-        if ([curgestureClass _fireGestures:gestures shouldCancelTouches:shouldCancelTouches]) {
-            if (DEBUG_GESTURES) {
+            // now test it against the other gestures, removing if we do not handle simultaneous gestures
+            for (UIGestureRecognizer* curGesture in g_currentlyTrackingGesturesList) {
```

Comment:
>curGesture [](start = 38, length = 10)

this name conflicts with the outer for loop; maybe call the outer one 'activeGesture' and the inner one 'otherGesture'?